### PR TITLE
Apply additional AssertJ best practices since #673

### PR DESCRIPTION
**Pull request by timtebeek**
## Issue
- Follows https://github.com/langchain4j/langchain4j/pull/673

## Change
Apply best practices once more, as time had passed between when #673 was opened and today. By clearing out any recipe suggestions on the main branch we ensure any new suggestions on PRs are limited to changed files.

## General checklist

- [x] There are no breaking changes
- [x] I have added unit and integration tests for my change
- [ ] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [ ] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)

### CODE REVIEW

**Comment by timtebeek** [comment]
I see failures here that I also see on another PR: 
- https://github.com/langchain4j/langchain4j/pull/1833
```
[INFO] Running dev.langchain4j.model.openai.OpenAiLanguageModelIT
Error:  Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 0.087 s <<< FAILURE! -- in dev.langchain4j.model.openai.OpenAiLanguageModelIT
Error:  dev.langchain4j.model.openai.OpenAiLanguageModelIT.should_use_enum_as_model_name -- Time elapsed: 0.014 s <<< ERROR!
java.lang.IllegalArgumentException: baseUrl cannot be null or empty
	at dev.ai4j.openai4j.OpenAiClient$Builder.baseUrl(OpenAiClient.java:178)
	at dev.langchain4j.model.openai.OpenAiLanguageModel.<init>(OpenAiLanguageModel.java:55)
	at dev.langchain4j.model.openai.OpenAiLanguageModel$OpenAiLanguageModelBuilder.build(OpenAiLanguageModel.java:38)
	at dev.langchain4j.model.openai.OpenAiLanguageModelIT.<init>(OpenAiLanguageModelIT.java:20)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.util.Optional.orElseGet(Optional.java:369)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
```

Not sure if that's something I can resolve, but it seems unrelated to the changes here.

File: langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/SchemaHelperTest.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ class Person {
 
         // then
         assertThat(schema.getRequiredList()).contains("name", "age", "isStudent", "friends");
-        assertThat(schema.getPropertiesMap().keySet()).contains("name", "age", "isStudent", "friends");
+        assertThat(schema.getPropertiesMap()).containsKey("name");
```
Comment (timtebeek) [inline langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/SchemaHelperTest.java]:
```suggestion
        assertThat(schema.getPropertiesMap()).containsKeys("name", "age", "isStudent", "friends");
```

File: langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/SchemaHelperTest.java [hunk 2]
```diff
@@ -76,7 +76,7 @@ void should_convert_json_schema_string_into_schema() {
 
         // then
         assertThat(schema.getRequiredList()).contains("artist-name", "artist-age", "artist-adult", "artist-pets", "artist-address");
-        assertThat(schema.getPropertiesMap().keySet()).contains("artist-name", "artist-age", "artist-adult", "artist-pets", "artist-address");
+        assertThat(schema.getPropertiesMap()).containsKey("artist-name");
```
Comment (timtebeek) [inline langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/SchemaHelperTest.java]:
```suggestion
        assertThat(schema.getPropertiesMap()).containsKeys("artist-name", "artist-age", "artist-adult", "artist-pets", "artist-address");
```

**Comment by timtebeek** [comment]
Can't see the details of the Snyk test failures; let me know if anything there is actionable.

**Comment by kpavlov** [comment]
@timtebeek , could you please sync your branch with the base branch?

**Review by ghost — APPROVED** [review]
@timtebeek thank you!

**Pull request merged**